src/main/java/de/tum/in/www1/artemis/service/iris/websocket/IrisWebsocketDTO.java

```diff
@@ -0,0 +1,77 @@
+package de.tum.in.www1.artemis.service.iris.websocket;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.iris.message.IrisMessage;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.status.PyrisStageDTO;
+import de.tum.in.www1.artemis.service.iris.IrisRateLimitService;
+import de.tum.in.www1.artemis.service.iris.exception.IrisException;
+
+/**
+ * A DTO for sending status updates of Iris to the client via the websocket
+ *
+ * @param type                the type of the message
+ * @param message             an IrisMessage instance if the type is MESSAGE
+ * @param errorMessage        the error message if the type is ERROR
+ * @param errorTranslationKey the translation key for the error message if the type is ERROR
+ * @param translationParams   the translation parameters for the error message if the type is ERROR
+ * @param rateLimitInfo       the rate limit information
+ * @param stages              the stages of the Pyris pipeline
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record IrisWebsocketDTO(IrisWebsocketMessageType type, IrisMessage message, String errorMessage, String errorTranslationKey, Map<String, Object> translationParams,
+        IrisRateLimitService.IrisRateLimitInformation rateLimitInfo, List<PyrisStageDTO> stages) {
+
+    /**
+     * Creates a new IrisWebsocketDTO instance with the given parameters
+     * Takes care of setting the type correctly and also extracts the error message and translation key from the throwable (if present)
+     *
+     * @param message       the IrisMessage (optional)
+     * @param throwable     the Throwable (optional)
+     * @param rateLimitInfo the rate limit information
+     * @param stages        the stages of the Pyris pipeline
+     * @return the created IrisWebsocketDTO instance
+     */
+    public static IrisWebsocketDTO create(IrisMessage message, Throwable throwable, IrisRateLimitService.IrisRateLimitInformation rateLimitInfo, List<PyrisStageDTO> stages) {
```

Comment:
why is this not declared as a normal constructor? I do not see a benefit of having a static create method